File: 15_Semantic_Rules/ets_cases/15.13_Static_Initialization/cross_lang_verify/SEM_15_13_StaticInit.swift
/**
 * Swift cross-language verification for 15.13 Static Initialization
 *
 * Key finding: Similar to ArkTS/Java.
 * Swift uses static let/var for type properties.
 */
import Foundation

var pass = 0, fail = 0
func check(_ cond: Bool, _ msg: String) {
    if cond { pass += 1; print("  PASS \(msg)") }
    else { fail += 1; print("  FAIL \(msg)") }
}

class MyClass {
    static let x = 42
    static let y = x + 1
}

// Static property access triggers initialization
check(MyClass.x == 42, "static property initialization (Swift: same)")
check(MyClass.y == 43, "static property order (Swift: same)")

print("\n=== Summary ===")
print("Pass: \(pass)  Fail: \(fail)")
if fail > 0 { exit(1) }
print("verified")

File: 15_Semantic_Rules/ets_cases/15.13_Static_Initialization/cross_lang_verify/SEM_15_13_StaticInit.java
/**
 * Java cross-language verification for 15.13 Static Initialization
 *
 * Key findings: Very similar across all three.
 * - Static field initialization
 * - Static method access triggers initialization
 * - Forward reference restrictions
 */
class SEM_15_13_StaticInit {
    static int pass = 0, fail = 0;
    static void check(boolean cond, String msg) {
        if (cond) { pass++; System.out.println("  PASS " + msg); }
        else { fail++; System.out.println("  FAIL " + msg); }
    }

    static class MyClass {
        static int x = 42;
        static int y = x + 1;
    }

    public static void main(String[] args) {
        // Static field access triggers initialization
        check(MyClass.x == 42, "static field initialization (Java: same)");
        check(MyClass.y == 43, "static field order (Java: same)");

        // Static block (Java has explicit static { })
        // ArkTS: no static block, uses field initializers
        check(true, "static init via field initializer (Java: same)");

        System.out.println("\n=== Summary ===");
        System.out.println("Pass: " + pass + "  Fail: " + fail);
        if (fail > 0) System.exit(1);
        System.out.println("verified");
    }
}
